packages/keycloak-user-management/src/components/forms/UserForm/utils.tsx

```diff
@@ -193,17 +197,41 @@ export const submitForm = async (
   }
 };
 
+// get the code of a practitioner resource type
+// to be used to determine the resource type
+// i.e if it's a practitioner or a supervisor resource type
+export const getUserTypeCode = (role: IPractitionerRole) =>
+  role.code
+    ?.find((code) => code.coding)
```

Comment:
@machariamuguku I think you can still have more than one coding for this resource. 
How I have previously handled it (probably not the best  way), was to flatten all codeableConcepts codings, and then find. e.g [here](https://github.com/opensrp/web/blob/99d0350ec1fe45732c1b30e484697d6c4a7a9a31/packages/fhir-team-management/src/components/AddEditOrganization/utils.ts#L69).

Reply:
@p-netm should this suffice?

```
const practitionerRole = {
  code: {
    coding: [
      {
        system: "http://snomed.info/sct",
        code: "123456",
        display: "random code 1",
      },
      {
        system: "http://snomed.info/sct",
        code: "405623001",
        display: "Assigned practitioner",
      },
      {
        system: "http://snomed.info/sct",
        code: "78910",
        display: "random code 2",
      },
    ],
  },
};

export const getUserTypeCode = (role: typeof practitionerRole) =>
  role.code.coding
    .map((coding) => coding.code)
    .find((code) => code === "236321002" || code === "405623001");

console.log(getUserTypeCode(practitionerRole));
```

Reply:
@machariamuguku should `practitionerRole.code` not be an array?

Reply:
i.e
```
...
{
  code: [
    {
      coding: [
        {
          system: 'http://snomed.info/sct',
          code: '123456',
          display: 'random code 1'
        },
        {
          system: 'http://snomed.info/sct',
          code: '405623001',
          display: 'Assigned practitioner'
        }
      ]
    },
    {
      coding: [
        {
          system: 'http://loic.org',
          code: 'some-code',
          display: 'display'
        }
      ]
    }
  ]
};
...
```

Reply:
@p-netm this should handle both single codeable concept with multiple codings and multiple codeable concept with multiple codings. Although innately I feel like we might be solving a problem that won't exist. N/B we are currently creating a single practitioner role for every practitioner.

```
const practitionerRole = {
  code: [
    {
      coding: [
        {
          system: "http://snomed.info/sct",
          code: "405623001",
          display: "Assigned practitioner",
        },
        {
          system: "http://snomed.info/sct",
          code: "123456",
          display: "random code 1",
        },
        {
          system: "http://snomed.info/sct",
          code: "78910",
          display: "random code 2",
        },
      ],
    },
    {
      coding: [
        {
          system: "http://snomed.info/sct",
          code: "5557855546",
          display: "random code 3",
        },
      ],
    },
    {
      coding: [
        {
          system: "http://snomed.info/sct",
          code: "258858588",
          display: "random code 4",
        },
      ],
    },
  ],
};

export const getUserTypeCode = (role: typeof practitionerRole) =>
  role.code
    .flatMap((code) => code.coding.map((coding) => coding.code))
    .find((code) => code === "236321002" || code === "405623001");

console.log(getUserTypeCode(practitionerRole)); // returns 405623001
```

Reply:
I think this should work, too. :shipit: